### PR TITLE
kots cli get versions

### PR DESCRIPTION
**Pull request by jeffreygolden**
#### What type of PR is this?
type::feature

#### What this PR does / why we need it:
This introduces a new KOTS CLI command to collect all versions for an application.

Sample Output

```json
./replicatedhq/kots/bin/kots get versions 10gbtest -n default
[
	{
		"versionLabel": "0.1.4",
		"sequence": 4,
		"createdOn": "2022-01-21T16:06:15.430124Z",
		"status": "failed",
		"deployedAt": "2022-01-24T22:04:23.11637Z",
		"source": "Config Change"
	},
	{
		"versionLabel": "0.1.4",
		"sequence": 3,
		"createdOn": "2022-01-21T16:05:17.201138Z",
		"status": "failed",
		"deployedAt": "2022-01-24T22:04:03.45562Z",
		"source": "Upstream Update"
	},
	{
		"versionLabel": "0.1.3",
		"sequence": 2,
		"createdOn": "2022-01-20T18:20:21.891078Z",
		"status": "pending",
		"deployedAt": null,
		"source": "Upstream Update"
	}
]
```

#### Does this PR introduce a user-facing change?
```release-note
Feature: Add a CLI command to collect all versions for an application (https://kots.io/kots-cli/get/versions/)
```

#### Does this PR require documentation?
Yes: https://github.com/replicatedhq/kots.io/pull/654

### CODE REVIEW

File: cmd/kots/cli/get-versions.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+func getVersionsCmd(cmd *cobra.Command, args []string) error {
+	v := viper.GetViper()
+
+	if len(args) == 0 {
+		cmd.Help()
+		os.Exit(1)
+	}
+
+	appSlug := args[0]
```
Comment (sgalsaleh) [inline cmd/kots/cli/get-versions.go:46]: i thought this was a flag? at least that's what the pr description show

File: cmd/kots/cli/get-versions.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	newReq.Header.Add("Content-Type", "application/json")
+	newReq.Header.Add("Authorization", authSlug)
+
+	resp, err := http.DefaultClient.Do(newReq)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to execute request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 500 {
```
Comment (sgalsaleh) [inline cmd/kots/cli/get-versions.go:139]: i would have this be
```
if resp.StatusCode != <expected-status-code> {
```
and the line below to be
```
return nil, errors.Errorf("unexpected status code %d", resp.StatusCode)
```

File: cmd/kots/cli/get-versions.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read")
+	}
+
+	appVersions := handlers.GetAppVersionsResponse{}
+	if err := json.Unmarshal(b, &appVersions); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal apps")
```
Comment (sgalsaleh) [inline cmd/kots/cli/get-versions.go:150]:
```suggestion
		return nil, errors.Wrap(err, "failed to unmarshal app versions")
```

File: cmd/kots/cli/get-versions.go (hunk 1 above)
Comment (jeffreygolden) [reply cmd/kots/cli/get-versions.go:46]: The docs align with not needing a flag here, and I've updated the PR description to match the code.

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**